node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1149 @@
+			)
+		)).await?;
+
+		Ok(rx.await?)
+	}
+
+	/// Logic that runs on startup.
+	async fn on_startup(&mut self) -> Result<(), Error> {
+		let validators = self.request_validators().await?;
+	
+		let roster = self.request_validator_groups().await?; 
+
+		self.table_context.key = signing_key(&validators[..], &self.keystore);
+
+		for assignment in roster.scheduled {
+			self.table_context.groups.insert(
+				assignment.para_id,
+				roster.validator_groups[assignment.group_idx.0 as usize]
+				.iter()
+				.map(|idx| validators[*idx as usize].clone())
+				.collect()
+			);
+		}
+
+		if let Some(ref key) = self.table_context.key {
+			for (para_id, group) in self.table_context.groups.iter() {
+				if group.contains(&key.as_ref().public()) {
+					self.assignment = *para_id;
+				}
+			}
+		}
+
+		self.validation_code = self.request_validation_code(self.assignment, 0, None).await?;
+		self.head_data = self.request_head_data(self.assignment).await?;
+		self.table_context.signing_context = self.request_signing_context().await?;
+		self.table_context.validators = validators;
+
+		Ok(())
+	}
+
+	/// Run asynchronously.
+	async fn run(mut self) -> Result<(), Error> {
+		self.on_startup().await?;
+
+		while let Some(msg) = self.rx_to.next().await {
+			match msg {
+				ToJob::CandidateBacking(msg) => {
+					self.process_msg(msg).await?;
+				}
+				_ => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	/// Validate the candidate that is requested to be `Second`ed and distribute validation result.
+	async fn validate_and_second(&mut self, candidate: AbridgedCandidateReceipt) -> Result<(), Error> {
+		let statement = match self.spawn_validation_work(candidate.clone()).await {
+			Ok(()) => {
+				// make PoV available for later distribution. Send data to the availability
+				// store to keep. Sign and dispatch `valid` statement to network if we
+				// have not seconded the given candidate.
+				Statement::Seconded(candidate)
+			}
+
+			// sign and dispatch `invalid` statement to network.
+			Err(_) => Statement::Invalid(candidate.hash()),
+		};
+
+		if let Some(ref signing_key) = self.table_context.key {
+			if let Some(local_index) = self.table_context.local_index() {
+				let signed_statement = SignedFullStatement::sign(
+					statement,
+					&self.table_context.signing_context,
+					local_index,
+					signing_key,
+				);
+
+				self.distribute_signed_statement(signed_statement).await?;
+			}
+		}
+
+		Ok(())
+	}
+
+	async fn import_statement(&mut self, statement: SignedFullStatement) -> Result<(), Error> {
+		let idx = statement.validator_index() as usize;
+
+		if self.table_context.validators.len() > idx {
+			match statement.check_signature(
+				&self.table_context.signing_context,
+				&self.table_context.validators[idx],
+			) {
+				Ok(()) => {
+					let stmt = primitive_statement_to_table(&statement);
+
+					self.table.import_statement(&self.table_context, stmt);
+
+					let proposed = self.table.proposed_candidates(&self.table_context);
```

Comment:
Most of the below logic seems redundant with what is already done by the `Summary` returned from `import_statement`.

Reply:
`backed_candidates` should contain all those candidates which have a majority of validity votes and nobody signaling them as bad. The `Summary` type is able to do that.

Reply:
However, the `RegisterBackingWatcher` interface is actually not that great because the statement table can withdraw something from the set of backed candidates after receiving an `Invalid` statement.